### PR TITLE
Remove some duplication from BulkRequestParser

### PR DESCRIPTION
**Pull request by original-brownbear**
No need to duplicate most of the index-request building 3x.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java [hunk 1]
```diff
@@ -341,58 +341,25 @@ public void parse(
...
+                            .versionType(versionType)
+                            .setPipeline(pipeline)
+                            .setIfSeqNo(ifSeqNo)
+                            .setIfPrimaryTerm(ifPrimaryTerm)
+                            .source(sliceTrimmingCarriageReturn(data, from, nextMarker, xContentType), xContentType)
+                            .setDynamicTemplates(dynamicTemplates)
+                            .setRequireAlias(requireAlias)
+                            .setRequireDataStream(requireDataStream)
+                            .setListExecutedPipelines(listExecutedPipelines);
+                        if ("create".equals(action)) {
```
Comment (kingherc) [inline server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java:357]:
```suggestion
                        if (opType != null && "create".equals(opType)) {
```

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java:357]: Not quite right? The old logic would set `true` if the action is "create" or if the op-type is create? Doing a null check before putting something on the right hand side of the equals seems unnecessary too?

**Comment by original-brownbear** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java (hunk 1 above)
Comment (kingherc) [reply server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java:357]: What I wanted to bring to your attention is that the current PR misses the path of the old code:
`"index".equals(action) -> opType != null -> .create("create".equals(opType))`. Right?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java:357]: Ah got it, that's actually covered 3 lines down in https://github.com/elastic/elasticsearch/pull/109302/files#diff-07b6912d7b05ace64d467c90faf865f31d682835a3b2c4d044c6a8db621f7481R360 

In this block we are either "index" or "create" so the `else if` is only taken for "index". That should replicate the old behavior 1:1 I think?

Comment (kingherc) [reply server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java:357]: Ah yes! Thanks for pointing out to me explicitly, for some reason I did not see it immediately.

**Review by kingherc — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Iraklis!

**Pull request merged**